### DISCUSSION TITLE
Add locale to the service container

### DISCUSSION THREAD

**Discussion by yamap55** [community/community#24923] [Actions]
<p>I used PostgreSQL service container.<br>
I want to set env to “LC_ALL=ja_JP.UTF-8”.<br>
but, the ja_JP.UTF-8 locale does not exist in the PostgreSQL service container.<br>
It is described like this in the docker file.</p>
<pre><code class="lang-auto">localedef -i ja_JP -c -f UTF-8 -A /usr/share/locale/locale.alias ja_JP.UTF-8
</code></pre>
<p>Please tell me how to settings.</p>
<h2>github actions yaml settings</h2>
<pre><code class="lang-yaml">jobs:
  pytest:
    runs-on: ubuntu-latest
    services:
      postgres:
        image: postgres:11.5
        env:
          POSTGRES_PASSWORD: postgres
          POSTGRES_USER: postgres
          POSTGRES_DB: test
          TZ: Asia/Tokyo
          LANG: ja_JP.UTF-8
          LANGUAGE: ja_JP:ja
          LC_ALL: ja_JP.UTF-8
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>I’ve looked at the “Locale Customization” section of the Docker Hub page for the <a href="https://hub.docker.com/_/postgres"><code>postgres</code> image</a>. The instructions there suggest to build your own container image with the locale you want.</p>
<p>To do that you will need a Dockerfile like this:</p>
<pre><code class="lang-auto">FROM postgres:11.5
RUN localedef -i ja_JP -c -f UTF-8 -A /usr/share/locale/locale.alias ja_JP.UTF-8
</code></pre>
<p>To use such an image as a service, you’d need to build it and push it to a public registry (for example <a href="https://docs.github.com/en/packages/guides/about-github-container-registry">GHCR</a>) and refer to it with the <code>image:</code> keyword of the service.</p>
<p>Alternatively, you could commit the Dockerfile to your repository and build the image during your Actions job and then start the service with <code>docker run</code>.</p>

**Comment by yamap55** [comment]
<p>thanks!</p>
<p>I understood that this could not be achieved with a service container.</p>
<p>I was able to run PostgreSQL with a github action.<br>
Make a note of the Dockerfile used and the yaml of GitHub Actions.</p>
<h2>Dockerfile</h2>
<pre><code class="lang-auto">FROM postgres:11.5
RUN localedef -i ja_JP -c -f UTF-8 -A /usr/share/locale/locale.alias ja_JP.UTF-8
</code></pre>
<h2>GitHub Actions setting file</h2>
<pre><code class="lang-yaml">name: postgresql

on: [push]

jobs:
  postgresql:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: PostgreSQL container build &amp; run
        run: |
          docker build -t postgresql_i .github/workflows
          docker run -d --name postgresql_t -p 5432:5432 -e POSTGRES_PASSWORD=postgres -e LC_ALL=ja_JP.UTF-8 postgresql_i
      - name: wait for db
        run: until docker exec postgresql_t pg_isready -U postgres; do sleep 1; done
      - name: show locale
        run: docker exec postgresql_t psql -U postgres -c "SHOW LC_COLLATE;"
</code></pre>
<h2>output</h2>
<pre><code class="lang-auto">Run docker exec postgresql_t psql -U postgres -c "SHOW LC_COLLATE;"
 lc_collate  
-------------
 ja_JP.UTF-8
(1 行)
</code></pre>
<h2>sample repository</h2>
<ul>
<li><a href="https://github.com/yamap55/github_actions_postgresql" class="inline-onebox" rel="noopener nofollow ugc">GitHub - yamap55/github_actions_postgresql</a></li>
</ul>

**Comment by airtower-luna** [comment]
<p>Nice, thank you for sharing your solution! :cat:</p>
